spec/controllers/welcome_controller_spec.rb

```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
+RSpec.describe WelcomeController, type: :controller do
+
+  describe 'GET index' do
+    let(:user_hash) do {
+      username: "Burp",
+      email_address: "thing@thing.com",
+      password: "this",
+      password_confirmation: "this"
+    }
```

Comment:
Have you learned about Factory Girl? It makes creating testing data sooo much easier.